cdap-ui/app/cdap/components/Replicator/Create/index.tsx

```diff
@@ -132,19 +138,23 @@ class CreateView extends React.PureComponent<ICreateProps, ICreateContext> {
   };
 
   public setNameDescription = (name, description) => {
-    this.setState({ name, description }, () => {
-      this.props.history.push(
-        `/ns/${getCurrentNamespace()}/replication/drafts/${this.state.draftId}`
-      );
-    });
+    this.setState({ name, description });
   };
 
   public setSourcePluginWidget = (sourcePluginWidget) => {
     this.setState({ sourcePluginWidget });
   };
 
   public setSourceConfig = (sourceConfig) => {
-    this.setState({ sourceConfig });
+    this.setState({ sourceConfig }, () => {
+      this.props.history.push(
```

Comment:
Is this mutating a `prop`?

Reply:
it's changing changing the history from React Router to update the URL